### PR TITLE
Implement breed-page and refactor service's logic

### PR DESCRIPTION
**Pull request by mgandore**

### CODE REVIEW

File: dogs/src/app/breed-profile/breed-profile.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<button routerLink="">All dogs</button>
+<div class="container">
+	<h1>{{dogName | titlecase}}</h1>
+	<div class="image-container">
+		<img src="{{imageURL}}" alt="Image of a dog">
+	</div>
+	<h2 *ngIf="subBreedNames.length > 0">{{dogName | titlecase}} sub-breeds</h2>
+	<h2 *ngIf="subBreedNames.length === 0">This breed has no sub-breeds :(</h2>
```
Comment (danielursu) [inline dogs/src/app/breed-profile/breed-profile.component.html]:
```suggestion
	<h2 *ngIf="subBreedNames.length > 0; esle elseBlock">{{dogName | titlecase}} sub-breeds</h2>
    <ng-template #elseBlock> 
        <h2>This breed has no sub-breeds :(</h2>
    </ng-template>
	
```

File: dogs/src/app/breed-profile/breed-profile.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	styleUrls: ["./breed-profile.component.css"]
+})
+export class BreedProfileComponent implements OnInit {
+
+	public dogName: string = "";
+	public imageURL: string = "";
+	public subBreedNames: string[] = [];
+	public responeseData: any;
+
+	public constructor(private apiService: ApiService, private route: ActivatedRoute) { }
```
Comment (danielursu) [inline dogs/src/app/breed-profile/breed-profile.component.ts]:
```suggestion
	public constructor(
	  private apiService: ApiService, 
	  private route: ActivatedRoute
	) { }
```

File: dogs/src/app/breed-profile/breed-profile.component.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			this.imageURL = this.responeseData.message;
+		});
+	}
+
+	private prepareDogSubBreed(): void {
+		this.apiService.getSubBreedNames(this.dogName).subscribe(data => {
+			this.responeseData = data;
+			const subBreedList = this.responeseData.message;
+			for (let subBreed of subBreedList) {
+				this.subBreedNames.push(subBreed);
```
Comment (danielursu) [inline dogs/src/app/breed-profile/breed-profile.component.ts]:
```suggestion
			for (let subBreedName of data.message) {
				this.subBreedNames.push(subBreedName);
```

File: dogs/src/app/breed-profile/breed-profile.component.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	private prepareDogName(): void {
+		this.route.paramMap.subscribe((params) => {
+			this.dogName = String(params.get("name"));
+		});
+	}
+
+	private prepareImageURL(): void {
+		this.apiService.getImageURL(this.dogName).subscribe(data => {
+			this.responeseData = data;
+			this.imageURL = this.responeseData.message;
```
Comment (danielursu) [inline dogs/src/app/breed-profile/breed-profile.component.ts]:
```suggestion
			this.imageURL = data.message;
```

File: dogs/src/app/dogs-list/dogs-list.component.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ import { ApiService } from "../shared/api.service";
 export class DogsListComponent implements OnInit {
 
 	public dogNames: string[] = [];
+	public responseData: any;
```
Comment (danielursu) [inline dogs/src/app/dogs-list/dogs-list.component.ts]:
```suggestion
```

File: dogs/src/app/dogs-list/dogs-list.component.ts [hunk 6]
```diff
@@ -18,6 +19,12 @@ export class DogsListComponent implements OnInit {
 	}
 
 	private prepareDogList(): void {
-		this.dogNames = this.apiService.getDogs()
+		this.apiService.getDogNames().subscribe(data => {
+			this.responseData = data;
+			const dogNamseList = this.responseData.message;
+			for (let dogName in dogNamseList) {
+				this.dogNames.push(dogName);
+			}
```
Comment (danielursu) [inline dogs/src/app/dogs-list/dogs-list.component.ts]:
```suggestion
			for (let dogName of data.message) {
				this.dogNames.push(dogName);
			}
```
does it work with for-in?

File: dogs/src/app/shared/api.service.ts [hunk 7]
```diff
@@ -1,43 +1,25 @@
 import { HttpClient } from "@angular/common/http";
 import { Injectable } from "@angular/core";
 
+import { Observable } from "rxjs";
```
Comment (danielursu) [inline dogs/src/app/shared/api.service.ts:3]: move near @angular imports

**Review by danielursu — CHANGES_REQUESTED** [review]

File: dogs/src/app/dogs-list/dogs-list.component.ts (hunk 6 above)
Comment (mgandore) [reply dogs/src/app/dogs-list/dogs-list.component.ts]: Apparently "for key in object" is the way to iterate through the properties of an object (each breed is a property of "message" object according to this specific API)

**Review by danielursu — APPROVED** [review]

**Pull request merged**